programs/validator-history/src/state.rs

```diff
@@ -185,7 +189,12 @@ impl ValidatorHistory {
     pub const MAX_ITEMS: usize = MAX_ITEMS;
     pub const SEED: &'static [u8] = b"validator-history";
 
-    pub fn set_mev_commission(&mut self, epoch: u16, commission: u16) -> Result<()> {
+    pub fn set_mev_commission(
+        &mut self,
+        epoch: u16,
+        commission: u16,
+        mev_earned: u64,
+    ) -> Result<()> {
         // check if entry exists for the epoch
```

Comment:
this logic is wrong. mentioned to @ebatsell in diff PR, but if you update epoch 500, then 501, then 500 you'll end up with two epoch 500 entries. the second 500 update should go back and find the first 500 entry

Reply:
yeah for sure I see the other PR

Im thinking if calling [self.history.epoch_range(epoch,epoch)](https://github.com/jito-foundation/stakenet/pull/12/files#diff-d658c99b6787b63bdf8cc47cb1dab02627e07efad1ccb54b44bc4898302c336eR184 ) from that PR here on the circbuf will fix the logic

Reply:
nvm didnt need that, i pushed up what i think should address it

i think the case you mention would fail the existing check bc the `last_mut()`.epoch would != 500

```
  if let Some(entry) = self.history.last_mut() {
            if entry.epoch == epoch { // i don't think this would be true
                entry.mev_commission = commission;}
  
```
but yeah, a subequent update would fail, so the entry data could be stale (?)

Reply:
i deleted my prev comment on this thread - since it was incorrect

you're right, with the existing logic, in the case where we try and edit an epoch seen prev, it will create another entry for it.

i went ahead and pushed up a fix for that

Reply:
high level:
* the CircBuf struct should have at most one entry per epoch, and they should be laid out in ascending order
* if for some reason an entry is missed, it's pretty messy to try to insert after the fact due to the need to shift every following epoch down by one. 
* a new entry should be appended if an entry doesn't exist for that epoch and it's greater than the most recent epoch (`.last()`)

therefore the update pseudocode for all of these `set_`s should be something like:
```
if epoch > circ_buf.last().epoch {
  insert new entry
} else {
   if let Some(entry) = circ_buf.find(epoch) {
     modify entry
   }
}
```

Reply:
yeah makes sense, should be taken care of for all the `set_`s now